bootstrap/ansible/playbook.yml

```diff
@@ -1,293 +1,513 @@
+
+            # TODO: Firewall tasks
+
+          when: flag_lrc_enabled
+
+      when: provisioning_tasks == true and (deployment == "prod" or deployment == "staging")
+      tags: provisioning
+
+    # Development only provisioning tasks.
+    - name: Run Development Provisioning Tasks
+      block:
+        # Don't do anything as there aren't any development-specific tasks yet.
+        - meta: noop
+      # Uncomment when tasks are added.
+      # when: provisioning_tasks == true and deployment == "dev"
+      tags: provisioning
+
+    # Non-provisioning tasks common to development and production.
+    # Meant to be re-run frequently.
+    - name: Run Common Tasks
+      block:
+
+        # Configure Apache.
+
+        - name: Create a WSGI configuration file from a template if SSL is disabled
+          template:
+            src: "{{ git_prefix }}/{{ reponame }}/bootstrap/ansible/wsgi_conf.tmpl"
+            dest: /etc/httpd/conf.d/{{ wsgi_conf_file_name }}
+          when: not ssl_enabled
+
+        - name: Create a WSGI configuration file from a template if SSL is enabled
+          template:
+            src: "{{ git_prefix }}/{{ reponame }}/bootstrap/ansible/wsgi_conf_ssl.tmpl"
+            dest: /etc/httpd/conf.d/{{ wsgi_conf_file_name }}
+          when: ssl_enabled
+
+        # Configure the Django application.
+
+        - name: Copy the ColdFront local_settings.py file from the sample
+          copy:
+            src: "{{ git_prefix }}/{{ reponame }}/coldfront/config/local_settings.py.sample"
+            dest: "{{ git_prefix }}/{{ reponame }}/coldfront/config/local_settings.py"
+
+        - name: Copy the ColdFront local_strings.py file from the sample
+          copy:
+            src: "{{ git_prefix }}/{{ reponame }}/coldfront/config/local_strings.py.sample"
+            dest: "{{ git_prefix }}/{{ reponame }}/coldfront/config/local_strings.py"
+
+        - name: Create the deployment-specific _settings.py file from a template
+          template:
+            src: "{{ git_prefix }}/{{ reponame }}/bootstrap/ansible/settings_template.tmpl"
+            dest: "{{ git_prefix }}/{{ reponame }}/{{ djangoprojname }}/config/{{ deployment }}_settings.py"
+
+        - name: Create the deployment-specific password reset email subject text file from a template
+          template:
+            src: "{{ git_prefix }}/{{ reponame }}/bootstrap/ansible/password_reset_subject_template.tmpl"
+            dest: "{{ git_prefix }}/{{ reponame }}/{{ djangoprojname }}/core/user/templates/user/passwords/password_reset_subject.txt"
+
+        # Restart Services.
+
+        - name: Restart and enable Redis service
+          service:
+            name: redis
+            state: restarted
+            enabled: yes
+
+        - name: Restart and enable Supervisor service
+          service:
+            name: supervisord
+            state: restarted
+            enabled: yes
+
+        - name: Restart and enable PostgreSQL service
```

Comment:
I wonder if we need to restart this each time?

Reply:
I wasn't sure if later tasks depended on using the restarted database so I included it after the initial postgres tasks just in case. I can remove the task if there arent any dependencies. Depending on the dependencies of the other restarting tasks, we may also be able to use Ansible "handlers", which only run when specified tasks return a "changed" output. Handlers can remove unnecessary restarting and speed up playbook execution.

Reply:
I don't think anything under "Run Common Tasks" requires that PostgreSQL be restarted first.

Can we perform the "enable" during provisioning?

Reply:
When provisioning is enabled, the common tasks are run (the conditional is `when: provisioning_tasks == true or common_tasks == true`) so enabling happens under provisioning. I'll remove the first restart.